src/Providers/OpenAI/Handlers/Stream.php

```diff
@@ -123,17 +147,49 @@ protected function parseNextDataLine(StreamInterface $stream): ?array
      */
     protected function extractToolCalls(array $data, array $toolCalls): array
     {
-        foreach (data_get($data, 'choices.0.delta.tool_calls', []) as $index => $toolCall) {
-            if ($name = data_get($toolCall, 'function.name')) {
-                $toolCalls[$index]['name'] = $name;
-                $toolCalls[$index]['arguments'] = '';
-                $toolCalls[$index]['id'] = data_get($toolCall, 'id');
+        $type = data_get($data, 'type', '');
+
+        if ($type === 'response.output_item.added' && data_get($data, 'item.type') === 'function_call') {
+            $index = (int) data_get($data, 'output_index', count($toolCalls));
+
+            $toolCalls[$index]['id'] = data_get($data, 'item.id');
+            $toolCalls[$index]['call_id'] = data_get($data, 'item.call_id');
+            $toolCalls[$index]['name'] = data_get($data, 'item.name');
+            $toolCalls[$index]['arguments'] = '';
+
+            if ($this->pendingReasoningId !== null) {
+                $toolCalls[$index]['reasoning_id'] = $this->pendingReasoningId;
+                $this->pendingReasoningId = null;
+            }
+
+            return $toolCalls;
+        }
+
+        if ($type === 'response.function_call_arguments.delta') {
+            $callId = data_get($data, 'item_id');
+            $delta = data_get($data, 'delta', '');
+
+            foreach ($toolCalls as &$call) {
+                if (($call['id'] ?? null) === $callId) {
+                    $call['arguments'] .= $delta;
+                    break;
+                }
             }
 
-            $arguments = data_get($toolCall, 'function.arguments');
+            return $toolCalls;
+        }
```

Comment:
Do you think this is necessary ?

as argument will be handled when `type='response.function_call_arguments.done'`